xwiki-rendering-macros/xwiki-rendering-macro-message/src/main/java/org/xwiki/rendering/internal/macro/message/MacroIconPrettyNameProvider.java

```diff
@@ -19,21 +19,18 @@
  */
 package org.xwiki.rendering.internal.macro.message;
 
-import org.xwiki.component.annotation.Component;
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
 
-import javax.inject.Singleton;
-import java.util.Map;
-
-@Component
-@Singleton
-class MacroIconPrettyNameProvider {
-  private static final Map<String, String> PRETTYNAME_MAPPING = Map.of(
-    "exclamation", "Error",
-    "information", "Information",
-    "accept", "Success",
-    "error", "Warning"
-  );
-  String getIconPrettyName(String iconName) {
-    return PRETTYNAME_MAPPING.get(iconName);
-  }
+/**
+ * Provide an alternative text for an icon.
+ *
+ * @version $Id$
+ * @since 16.7.0RC1
+ */
+@Role
+@Unstable
+interface MacroIconPrettyNameProvider 
```

Comment:
Honestly, you don't need to go through an interface for an internal component. Your previous one was fine, was just missing a role (you can set itself as role in the `@Component`).